addons/website_sale_picking/models/website.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class Website(models.Model):
+    _inherit = 'website'
+
+    picking_site_ids = fields.Many2many('delivery.carrier', string='Picking sites',
+                                        compute='_compute_picking_sites', store=False)
+
+    def _compute_picking_sites(self):
+        for record in self:
+            record.picking_site_ids = self.env['delivery.carrier'].search([('website_id.id', '=', record.id)])
```

Comment:
Shouldn't this also include delivery carriers that don't have a `website_id`?
Also please do this in a single search, either by doing a `_read_group` or by doing a `search` followed by grouping on `website_id`, I prefer the latter.

Reply:
> Shouldn't this also include delivery carriers that don't have a website_id?
It depends if you want to see them explicitly in the website config or only in the delivery carriers page. I thought it was more logical to edit common delivery carriers in their page to avoid changing delivery carriers for multiple websites without noticing it